browser/ui/tabs/tab_menu_model_delegate_proxy.cc

```diff
@@ -0,0 +1,154 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/tabs/tab_menu_model_delegate_proxy.h"
+
+#include "base/auto_reset.h"
+#include "brave/browser/ui/tabs/brave_tab_menu_model.h"
+#include "chrome/app/chrome_command_ids.h"
+#include "chrome/browser/defaults.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/sessions/tab_restore_service_factory.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_commands.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "chrome/browser/ui/tabs/tab_strip_model_delegate.h"
+#include "chrome/grit/generated_resources.h"
+#include "components/sessions/core/tab_restore_service.h"
+
+// static
+// Temporarily store pointer before BraveTabMenuModel takes ownership.
+// Then, this global pointer becomes null by BraveTabMenuModel.
+// It looks weird but can't avoid because we should create this and pass to
+// |TabMenuModel| while ctoring |BraveTabMenuModel|.
+// I think there is no way to initialze member variables with params before
+// initializing base class.
+// Because of that, storing it in a global pointer and then |BraveTabMenuModel|
+// takes its ownership.
+// With this we can add our items to tab context menu w/o patching.
+ui::SimpleMenuModel::Delegate* TabMenuModelDelegateProxy::s_delegate_proxy_;
+
+// static
+ui::SimpleMenuModel::Delegate* TabMenuModelDelegateProxy::GetInstance(
+    ui::SimpleMenuModel::Delegate* delegate,
+    TabStripModel* tab_strip,
+    int index) {
+  // |s_delegate_proxy_| should be always null here because
+  // BraveTabMenuModel takes its ownership in its ctor.
+  DCHECK(!s_delegate_proxy_);
+  s_delegate_proxy_ = new TabMenuModelDelegateProxy(delegate, tab_strip, index);
```

Comment:
`TabMenuModel`’s parent is `ui::SimpleMenuModel`, which will keep the delegate reference
so you can just return it, I think (without storing it) 
(unless the static reference is being used for testing)

Reply:
Oh, I didn't know about SimpleMenuModel::delegate()!!
|s_delegate_proxy_| is deleted. 👍